components/markdown/src/shortcode/parser.rs

```diff
@@ -1,27 +1,46 @@
-use std::ops::Range;
+use std::{collections::HashMap, ops::Range};
 
 use errors::{bail, Context as ErrorContext, Result};
 use libs::tera::{to_value, Context, Map, Tera, Value};
 use pest::iterators::Pair;
 use pest::Parser;
 use pest_derive::Parser;
-use std::collections::HashMap;
-use utils::templates::ShortcodeFileType;
+use utils::templates::{ShortcodeDefinition, ShortcodeFileType, ShortcodeInvocationCounter};
 
 pub const SHORTCODE_PLACEHOLDER: &str = "@@ZOLA_SC_PLACEHOLDER@@";
 
 #[derive(PartialEq, Debug, Eq)]
 pub struct Shortcode {
     pub(crate) name: String,
     pub(crate) args: Value,
+    // In practice, span.len() is always equal to SHORTCODE_PLACEHOLDER.len()
     pub(crate) span: Range<usize>,
     pub(crate) body: Option<String>,
     pub(crate) nth: usize,
+    pub(crate) inner: Vec<Shortcode>,
     // set later down the line, for quick access without needing the definitions
     pub(crate) tera_name: String,
 }
 
 impl Shortcode {
+    /// Attempts to fill the `tera_name` field from the provided definitions for self and all of self.inner.
```

Comment:
`shortcodes/xxx.yy`. This function re-implements what was originally [here](https://github.com/getzola/zola/pull/2748/files/4b39a97bd715a76326a6fca883272d7263b70472#diff-82dc1d519d7c352b36650805f58639cec80b5aac264cc565b3919b1d78556814L19-R20) so it may be done recursively with inner shortcodes